### PR TITLE
Treat staging repos as authoritative source for all files

### PR DESCRIPTION
**Pull request by spiffxp**
Add CONTRIBUTING.md files to remind people not to PR directly into the published repos.

/hold
I believe this requires removing these files from the published repos if any copies exist there

ref: kubernetes/kubernetes#57559

### CODE REVIEW

**Comment by sttts** [comment]
Can you separate the moved client-go files into a nother PR? They need further discussion, compare https://github.com/kubernetes/kubernetes/issues/57559#issuecomment-353682964.

File: staging/src/k8s.io/client-go/.github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Sorry, client-go does not accept changes via pull requests at this time. Please
```
Comment (sttts) [inline staging/src/k8s.io/client-go/.github/PULL_REQUEST_TEMPLATE.md]: Should we add this file everywhere?

File: staging/src/k8s.io/client-go/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,307 @@
+TODO: This document was manually maintained so might be incomplete. The
```
Comment (sttts) [inline staging/src/k8s.io/client-go/CHANGELOG.md]: this file can only move if we agree to integrate the client-go release process into the kube release process. This definitely needs further discussion. If we want to go this route, we need a published-repos roles in the release team with the authority to do whatever is needed during a release.

File: staging/src/k8s.io/client-go/.github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (spiffxp) [reply staging/src/k8s.io/client-go/.github/PULL_REQUEST_TEMPLATE.md]: good idea, I'll take another pass at this

File: staging/src/k8s.io/client-go/CHANGELOG.md (hunk 2 above)
Comment (spiffxp) [reply staging/src/k8s.io/client-go/CHANGELOG.md]: Are there use cases for this that would require direct write access to kubernetes/kubernetes?
 If so, I would really appreciate it if you could articulate them here: https://github.com/kubernetes/kubernetes/issues/57667

Comment (sttts) [reply staging/src/k8s.io/client-go/CHANGELOG.md]: It's about agility during tagging of client-go, not so much about write access. Every change in k/k to show up in client-go needs a publishing-bot run (every few hours right now).

Comment (sttts) [reply staging/src/k8s.io/client-go/CHANGELOG.md]: And the client-go tags happen asynchronously to kube right now. Which means that we would want to update k/k branches after the official tags. This involves the release (branch) manager to agree on those.

**Comment by spiffxp** [comment]
rebased

File: staging/src/k8s.io/apiserver/.github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Sorry, we do not accept changes directly against this repository. Please see
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/.github/PULL_REQUEST_TEMPLATE.md]: there must be removed from client-go first

**Comment by spiffxp** [comment]
Looks like I need to get `verify-generated-protobuf.sh` to play nicely
```
I0105 20:16:25.634] --- /go/src/k8s.io/kubernetes/staging/src/k8s.io/api/.github/PULL_REQUEST_TEMPLATE.md	2018-01-05 19:59:02.884211968 +0000
I0105 20:16:25.635] +++ /go/src/k8s.io/kubernetes/_tmp/staging/src/k8s.io/api/.github/PULL_REQUEST_TEMPLATE.md	1970-01-01 00:00:00.000000000 +0000
I0105 20:16:25.635] @@ -1,2 +0,0 @@
I0105 20:16:25.635] -Sorry, we do not accept changes directly against this repository. Please see 
I0105 20:16:25.635] -CONTRIBUTING.md for information on where and how to contribute instead.
I0105 20:16:25.697] staging/src/k8s.io/api is out of date. Please run hack/update-generated-protobuf.sh
I0105 20:16:25.759] FAILED   verify-generated-protobuf.sh	74s
```

**Review by caesarxuchao — COMMENTED** [review]
lgtm except a nit.

**Comment by sttts** [comment]
/unhold

@spiffxp If you still have your bash command around to check for duplicated files in the repos and staging/, would be worth to check whether we forgot to remove any file.

**Comment by sttts** [comment]
/approve
/assign @thockin 

@thockin we need a top-level approval as this touches all staging repos.

**Comment by sttts** [comment]
@liggitt @deads2k @DirectXMan12 can you approve?

**Comment by deads2k** [comment]
/approve no-issue

consider a squash

**Comment by spiffxp** [comment]
@deads2k good point, squash-rebased

**Comment by sttts** [comment]
@liggitt @DirectXMan12 we need you for an approval

**Comment by liggitt** [comment]
/approve
change to api paths were not api-related

**Pull request merged**